confdb/confdb.go

```diff
@@ -1300,33 +1463,62 @@ func (v *View) Get(databag Databag, request string) (any, error) {
 func mergeNamespaces(old, new any) (any, error) {
 	if old == nil {
 		return new, nil
+	} else if new == nil {
+		return old, nil
 	}
 
 	oldType, newType := reflect.TypeOf(old).Kind(), reflect.TypeOf(new).Kind()
 	if oldType != newType {
 		return nil, fmt.Errorf("cannot merge results of different types %T, %T", old, new)
 	}
 
-	if oldType != reflect.Map {
-		// if the values are both scalars/lists, the new replaces the old value
+	if oldType != reflect.Map && oldType != reflect.Slice {
+		// if the values are both scalars, the new value replaces the old one
 		return new, nil
 	}
 
-	// if the values are maps, merge them recursively
-	oldMap, newMap := old.(map[string]any), new.(map[string]any)
-	for k, v := range newMap {
-		if storeVal, ok := oldMap[k]; ok {
+	if oldType == reflect.Map {
+		oldMap, newMap := old.(map[string]any), new.(map[string]any)
+		return mergeMaps(oldMap, newMap)
+	}
+
+	oldList, newList := old.([]any), new.([]any)
+	return mergeLists(oldList, newList)
+}
+
+func mergeMaps(old, new map[string]any) (map[string]any, error) {
+	for k, v := range new {
+		if storeVal, ok := old[k]; ok {
 			merged, err := mergeNamespaces(storeVal, v)
 			if err != nil {
 				return nil, err
 			}
 			v = merged
 		}
 
-		oldMap[k] = v
+		old[k] = v
+	}
+
+	return old, nil
+}
+
+func mergeLists(old, new []any) ([]any, error) {
```

Comment:
it would be could for all these merge* helpers to have description of what they do,  merging of maps is kind of straightforward but not that list are in the pictures not so much